### ISSUE TITLE
Bug in inequality operator of Option in Version 5.0.0

### ISSUE THREAD

**Issue by BADF00D** [Bomret/NeverNull#17]
Hi,
I just stumbled upon an error.
```csharp
var x = Option.From(1);
var y = Option.From(2);

Console.WriteLine(x != y); // outputs false
```

The problem is, that the implementation of the inequality operator is exactly the same as the equality operator:
```csharp
public static bool operator ==(Option<T> left, Option<T> right) =>
            ((IStructuralEquatable)left).Equals(right, EqualityComparer<object>.Default);

public static bool operator !=(Option<T> left, Option<T> right) =>
            ((IStructuralEquatable)left).Equals(right, EqualityComparer<object>.Default);
```

The inequality operator should be something like:
```csharp
public static bool operator !=(Option<T> left, Option<T> right) =>
            !((IStructuralEquatable)left).Equals(right, EqualityComparer<object>.Default);

//or
public static bool operator !=(Option<T> left, Option<T> right) => !(left == right);
```

Because there were no tests for verification, I would like to add some:
```csharp
[TestFixture]
class If_Equality_operator_is_called {
     [Test]
    public bool Result_should_be_correct(Option<int> x, Option<int> y){
        return x == y;
    }
    private static IEnumerable<TestCaseData> EqulityOperatorTestCases{
        get{
            return TestCaseData(Option<int>.None, Option.None).Returns(true);
            return TestCaseData(Option<int>.None, Option.From(1)).Returns(false);
            return TestCaseData(Option.From(1), Option<int>.None).Returns(false);
            return TestCaseData(Option.From(1), Option.From(2)).Returns(false);
            return TestCaseData(Option.From(1), Option.From(1)).Returns(true);
        }
    }
}
[TestFixture]
class If_inequality_operator_is_called {
     [Test]
    public bool Result_should_be_correct(Option<int> x, Option<int> y){
        return x != y;
    }
    private static IEnumerable<TestCaseData> EqulityOperatorTestCases{
        get{
            return TestCaseData(Option<int>.None, Option.None).Returns(false);
            return TestCaseData(Option<int>.None, Option.From(1)).Returns(true);
            return TestCaseData(Option.From(1), Option<int>.None).Returns(true);
            return TestCaseData(Option.From(1), Option.From(2)).Returns(true);
            return TestCaseData(Option.From(1), Option.From(1)).Returns(false);
        }
    }
}
```

I would have created a pull request, but unfortunately I was not able to build the project.

**Comment by Bomret** [comment]
Ahh, damn! Good catch! I’ll try to fix this ASAP.

**Comment by BADF00D** [comment]
I saw you fixed the issue, but forgot the release NuGet package 5.0.1.

**Comment by Bomret** [comment]
Take a look at the time of the latest commits. I did not forget it 😉

**Comment by Bomret** [comment]
fixed in 5.1.0
